sdk/openai/openai/package.json

```diff
@@ -95,6 +97,7 @@
     "@microsoft/api-extractor": "^7.31.1",
     "@types/mocha": "^7.0.2",
     "@types/node": "^14.0.0",
+    "@typespec/compiler": "^0.45.0",
```

Comment:
We haven't been depending on codegen tools in the past and rely on global install (`autorest` for example), since re-generating is not done very often and most likely not by community contributors.  Not against it either but want to start a discussion. It may have benefit like pinning to a specific version.

Reply:
Good point. My intention here is to make it easier for our partners to contribute to the library especially that I will leave on vacation for a month. I added `rushx generate` which generates using the new DPG and you can see it is not very straightforward. I hope DPG will evolve with a simpler API and then, we can remove that convenience command.

Reply:
We should discuss the expected end-to-end workflow before adding `@typespec/compiler` here.  We also want the end-to-end to be as similar as possible across all language repos.

Reply:
@mikeharder The end-to-end workflow is simply to re-generate the SDK using the JS DPG which is still in early days. If you take a look at the `generate` script in package.json, you will see that it is quite complex to justify its own script. And of course the script will not work without having the typespec compiler installed.

The alternative is to move the script into a bash script with the assumption that the user has to install the compiler manually but the issue is that it will probably not work cross-platform and adds one extra hoop.

Let me know if you have any good suggestions to follow here.

Reply:
Discussed offline. Removed the dep in https://github.com/Azure/azure-sdk-for-js/pull/26138/commits/f1e2101f6073683c9bf82002c89d3674b8a1b7a4